src/lib/alloc.c

```diff
@@ -234,11 +234,18 @@ static void *alloc_cont_blocks(struct mm_heap *heap, int level,
 	/* check if we have enough consecutive blocks for requested
 	 * allocation size.
 	 */
-	for (current = map->first_free; current < map->count; current++) {
+	for (current = map->first_free; current < map->count &&
+	     remaining < count; current++) {
 		hdr = &map->block[current];
 
-		if (!hdr->used)
+		if (hdr->used) {
+			/* not suitable, reset */
+			remaining = 0;
+		} else {
+			if (!remaining)
```

Comment:
@keyonjie, wouldn't remove this `if(!remaining)` and move the assignment `start = current` below `remaining = 0` be better?

Reply:
@mrajwa no, the logic here is different, we set start at (!hdr->used), not at (hdr->used), get it?

Reply:
Yes, I assumed `start = current + 1;` but then it would be nice to check if this newly calculated start has not exceeded the amount of blocks we have. Although in theory even if we went above the limit it will not cause any issue because the next interaction of for loop will stop it end we will end up with `remaining < count ` therefore terminating the whole allocation.

Reply:
```
if (hdr->used) {
			/* not suitable, reset */
			remaining = 0;
                        start = current + 1; /* new start */
		} else {	
			remaining++;
		}
```

and we don't care if `start > amount_of_blocks`

Reply:
> ```
> if (hdr->used) {
>     /* not suitable, reset */
>     remaining = 0;
>     start = current + 1; /* new start */

But the current + 1 still could be "used", the current here make no sense for us.

> } else {	
>     remaining++;
> }
> ```
> 
> and we don't care if `start > amount_of_blocks`

Reply:
Yes the `current + 1` does not guarantee free block. If it indeed is not free then it will be updated in the next iteration.

Reply:
That's why I don't do it there, assigning start blindly is in low efficiency.